core/src/main/java/io/grpc/transport/WritableBufferAllocator.java

```diff
@@ -32,13 +32,14 @@
 package io.grpc.transport;
 
 /**
- * The preferred way of creating a {@link WritableBuffer}.
+ * An allocator of buffers provided by the transport implementation to {@link MessageFramer} so
+ * it can send chunks of data to the transport in a form that the transport can directly serialize.
  */
 public interface WritableBufferAllocator {
 
   /**
-   * Allocate a new {@link WritableBuffer} that can hold
-   * {@code capacity} bytes.
+   * Request a new {@link WritableBuffer} with the given {@code capacity}. The allocator is
+   * free to return a buffer with a greater or lesser capacity.
```

Comment:
I'd favor 4k over 16k for the minimum size. We expect many applications to use messages < 4k exclusively, so choosing 16k would just cause waste. Stopping at 1 MB sounds fine.

Why is there such a high variation between the two sets of tests:
ByteBufAllocatorBenchmark.defaultPooledDirectAllocAndFree   65536  thrpt   20   2409135.087 ± 136903.037  ops/s
ByteBufAllocatorBenchmark.defaultPooledDirectAllocAndFree   0065536  thrpt   20  3024966.732 ±  83142.423  ops/s